packages/blade/src/components/Form/FormHintText.tsx

```diff
@@ -9,15 +9,48 @@ import { getPlatformType } from '~utils';
 import Box from '~components/Box';
 
 type FormHintTextProps = {
-  id?: string;
-  children: React.ReactNode;
-  variant: 'help' | 'error' | 'success';
+  state: 'help' | 'error' | 'success';
+  /**
+   * Help text for the group
+   */
+  helpText?: string;
+  /**
+   * Error text for the group
+   *
+   * Renders when `state` is set to 'error'
+   */
+  errorText?: string;
+  /**
+   * Success text for the group
+   *
+   * Renders when `state` is set to 'success'
+   */
+  successText?: string;
+  /**
+   * Sets the id on errorText.
+   * Needed for accessibility reasons.
+   */
+  errorTextId?: string;
+  /**
+   * Sets the id on helpText.
+   * Needed for accessibility reasons.
+   */
+  helpTextId?: string;
+  /**
+   * Sets the id on successText.
+   * Needed for accessibility reasons.
+   */
+  successTextId?: string;
 };
 
 const FormHintText = ({
-  id,
-  children,
-  variant = 'help',
+  state,
+  errorText,
+  successText,
+  helpText,
+  helpTextId,
```

Comment:
why do we need ids? for a11y?

Reply:
Yes

Reply:
what exactly does the ids do? we don't write logic for them right?